### PR TITLE
Add daemonset pkg unit tests

### PR DESCRIPTION
**Pull request by sebrandon1**
Resolves: #464 

Builds upon changes in #355 and #379 as well.

### CODE REVIEW

File: pkg/daemonset/daemonset.go [hunk 1]
```diff
@@ -66,29 +73,35 @@ func NewBuilder(
...
+		return builder
 	}
 
-	return &builder
+	return builder
 }
 
 // Pull loads an existing daemonSet into the Builder struct.
 func Pull(apiClient *clients.Settings, name, nsname string) (*Builder, error) {
 	glog.V(100).Infof("Pulling existing daemonset name:%s under namespace:%s", name, nsname)
```
Comment (kononovn) [inline pkg/daemonset/daemonset.go:102]: Please add verification that client is not nil.

File: pkg/daemonset/daemonset_test.go [hunk 2]
```diff
@@ -5,13 +5,20 @@ import (
...
+	appsv1 "k8s.io/api/apps/v1"
 	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	k8sfake "k8s.io/client-go/kubernetes/fake"
 )
 
-// buildValidTestBuilder returns a valid Builder for testing purposes.
-func buildValidTestBuilder() *Builder {
+func buildValidTestBuilderWithClient(objects []runtime.Object) *Builder {
```
Comment (kononovn) [inline pkg/daemonset/daemonset_test.go:15]: Please move all private functions to the end of the file

File: pkg/daemonset/daemonset_test.go [hunk 3]
```diff
@@ -52,3 +65,391 @@ func TestWithOptions(t *testing.T) {
...
+			},
+		}
+	}
+
+	testCases := []struct {
+		name                string
+		namespace           string
+		expectedError       bool
+		addToRuntimeObjects bool
+		expectedErrorText   string
```
Comment (kononovn) [inline pkg/daemonset/daemonset_test.go:171]: Add client and verify that Pull returns error if apiClient is nil.

**Review by kononovn — CHANGES_REQUESTED** [review]
Few comments to fix

File: pkg/daemonset/daemonset_test.go [hunk 4]
```diff
@@ -52,3 +65,391 @@ func TestWithOptions(t *testing.T) {
...
+				Name:      name,
+				Namespace: namespace,
+			},
+		}
+	}
+
+	testCases := []struct {
+		name                string
+		namespace           string
+		expectedError       bool
```
[the remark this answers is not in the document]
Comment (sebrandon1) [reply pkg/daemonset/daemonset_test.go:155]: Done.

File: pkg/daemonset/daemonset_test.go (hunk 3 above)
Comment (sebrandon1) [reply pkg/daemonset/daemonset_test.go:156]: Done.

**Review by kononovn — APPROVED** [review]
/lgtm

**Pull request merged**